dlt/common/configuration/container.py

```diff
@@ -40,48 +55,112 @@ def __getitem__(self, spec: Type[TConfiguration]) -> TConfiguration:
         if not issubclass(spec, ContainerInjectableContext):
             raise KeyError(f"{spec.__name__} is not a context")
 
-        item = self.contexts.get(spec)
+        context, item = self._thread_getitem(spec)
         if item is None:
             if spec.can_create_default:
                 item = spec()
-                self.contexts[spec] = item
+                self._thread_setitem(context, spec, item)
                 item.add_extras()
             else:
                 raise ContextDefaultCannotBeCreated(spec)
 
-        return item  # type: ignore
+        return item  # type: ignore[return-value]
 
     def __setitem__(self, spec: Type[TConfiguration], value: TConfiguration) -> None:
         # value passed to container must be final
         value.resolve()
         # put it into context
-        self.contexts[spec] = value
+        self._thread_setitem(self._thread_context(spec), spec, value)
 
     def __delitem__(self, spec: Type[TConfiguration]) -> None:
-        del self.contexts[spec]
+        context = self._thread_context(spec)
+        self._thread_delitem(context, spec)
 
     def __contains__(self, spec: Type[TConfiguration]) -> bool:
-        return spec in self.contexts
+        context = self._thread_context(spec)
+        return spec in context
+
+    def _thread_context(
```

Comment:
can't you use pythons thread local context to do all this? https://docs.python.org/3/library/threading.html#thread-local-data

Reply:
yeah I know it but when you look at the code, there are exceptions to that behavior. 
1. some type of context are available globally (I use main thread id)
2. there's a special treatment of the executor thread pool. I use a context of a thread that started a pool, not the current thread

so yeah I could use `local()` but there are exceptions so I'd need to keep more dictionaries. or you can force the thread id for local()?